ovirtlib4/system_service.py

```diff
@@ -64,8 +70,121 @@ def _get_collection_entity(self):
         """
         return CollectionEntity(connection=self.connection)
 
-    def get(self, *args, **kwargs):
-        """ Same as list() """
+    def _get_wait_for_type(self, wait_for):
+        """
+        Detect the given 'wait_for' type
+
+        Args:
+            wait_for (bool/callable/str/None): object from type of string, function name or boolean
+
+        Returns:
+            bool/callable/str: If the type of given 'wait_for' if detected, None otherwise
+        """
+        try:
+            if type(wait_for) == bool:
+                return bool
+        except Exception:
+            pass
+
+        try:
+            if callable(wait_for):
+                return callable
+        except Exception:
+            pass
+
+        try:
+            if type(wait_for) == str:
+                return str
+        except Exception:
+            pass
+
+        return None
+
+    def _process_sample(self, wait_for, wait_for_method, sample):
+        """
+        Handle a result sample according to the given 'wait_for' and 'wait_for_method'
+
+        Args:
+            wait_for (object): see run_sampler() docstring
+            wait_for_method (str): see run_sampler() docstring
+            sample (list): List of SDK entities, returned by get()
+
+        Returns:
+            list: the given sample if it match, otherwise an empty list
+        """
+        wait_for_type = self._get_wait_for_type(wait_for=wait_for)
```

Comment:
maybe you can give up on the method _get_wait_for_type because in any way you chack the same here so you can do it here like:
        if type(wait_for) == bool:
...
        if type(wait_for) == str:
...
        if callable(wait_for):